modelling/src/test/java/org/axonframework/modelling/AnnotationBasedEntityEvolvingComponentTest.java

```diff
@@ -357,4 +363,97 @@ void handle(String event) {
             this.handledCount++;
         }
     }
+
+    @Nested
+    class PolymorphicEntitySupport {
+
+        // Sealed interface where the ENTITY ITSELF is polymorphic (not state inside)
```

Comment:
Brainfart: Perhaps we could call that a message-handling-entity? And the other a state-focused-entity? This isn't a comment that requires change anywhere. Just trying to discuss these different option we now have with AF5 (that may eventually land in the documentation).

Reply:
We essentially have:
1. Plan entity state
2. Event Sourced entities, state only
3. State-based entities with Command Handlers only
4. Event Sourced entities with command handlers

Option 1 and 2 are new to Axon Framework in it's current form. 3 and 4 already existed.